### PR TITLE
Devex - refactor work queue endpoints

### PR DESCRIPTION
**Pull request by codyseibert**
new endpoints include:

- v1/users/{userId}/messages/inbox
- v1/users/{userId}/messages/sent
- v1/sections/{section}/messages/inbox
- v1/sections/{section}/messages/sent
- v1/users/{userId}/document-qc/inbox
- v1/users/{userId}/document-qc/batched
- v1/users/{userId}/document-qc/served
- v1/sections/{section}/document-qc/inbox
- v1/sections/{section}/document-qc/batched
- v1/sections/{section}/document-qc/served

we will probably need a /processed endpoint for docket clerks

### CODE REVIEW

File: shared/src/business/useCases/workitems/getDocumentQCInboxForSectionInteractor.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}) => {
+  const user = applicationContext.getCurrentUser();
+
+  if (!isAuthorized(user, WORKITEM, user.userId)) {
+    throw new UnauthorizedError(
+      'Unauthorized for getting completed work items',
+    );
+  }
+
+  const sectionToExcept =
```
Comment (rachaelparris) [inline shared/src/business/useCases/workitems/getDocumentQCInboxForSectionInteractor.js:28]: What's this line supposed to do? If the user's section is senior attorney, set it to docket?

File: shared/src/business/test/completeWorkItemInteractor.e2e.test.js [hunk 2]
```diff
@@ -109,39 +111,11 @@ describe('completeWorkItem integration test', () => {
...
-            toUserId: '3805d1ab-18d0-43ec-bafb-654e83405416',
-          },
-        ],
-        section: 'petitions',
-        sentBy: 'richard',
-        sentBySection: 'petitions',
-        sentByUserId: '3805d1ab-18d0-43ec-bafb-654e83405416',
-      },
-    ]);
+    expect(outbox).toMatchObject([]);
```
Comment (rachaelparris) [inline shared/src/business/test/completeWorkItemInteractor.e2e.test.js:118]: Should this actually be an empty array? Completed messages don't go into the sent box?

File: shared/src/business/useCases/workitems/getDocumentQCInboxForSectionInteractor.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/business/useCases/workitems/getDocumentQCInboxForSectionInteractor.js:28]: yeah, the senior attorney is supposed to see the same records that the docketclerk sees

File: shared/src/business/test/completeWorkItemInteractor.e2e.test.js (hunk 2 above)
Comment (codyseibert) [reply shared/src/business/test/completeWorkItemInteractor.e2e.test.js:118]: I'll need to double check, but I think there was a requirement stating completed items do not show up in the sent box...

File: web-client/src/presenter/actions/chooseWorkQueueAction.js [hunk 3]
```diff
@@ -26,7 +26,70 @@ export const chooseWorkQueueAction = ({ get, path, props, store }) => {
   }
 
   let queuePrefs = get(state.workQueueToDisplay);
-  const workQueuePath = `${queuePrefs.queue}${queuePrefs.box}`;
+  let workQueueIsInternal = get(state.workQueueIsInternal);
+  let workQueuePath = `${workQueueIsInternal ? 'messages' : 'documentqc'}${
+    queuePrefs.queue
+  }${queuePrefs.box}`;
+
+  // if (workQueueIsInternal && queuePrefs.queue === 'my') {
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/chooseWorkQueueAction.js]: Can you delete these lines if they're not needed?

File: web-client/src/presenter/actions/getDocumentQCBatchedForSectionAction.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/**
+ * action for fetching all the work items associated with a user account.
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/getDocumentQCBatchedForSectionAction.js]: Update jsdoc? For this and the other actions below.

File: shared/src/business/test/completeWorkItemInteractor.e2e.test.js (hunk 2 above)
Comment (codyseibert) [reply shared/src/business/test/completeWorkItemInteractor.e2e.test.js:118]: Sarah has confirmed that when we "complete" a work item via that ui tab, no message is created.

**Review by rachaelparris — APPROVED** [review]

File: shared/src/business/useCases/workitems/getInboxMessagesForSectionInteractor.js [hunk 5]
```diff
@@ -5,24 +5,25 @@ const {
 const { UnauthorizedError } = require('../../../errors/errors');
 
 /**
+ * getInboxMessagesForSection
```
Comment (matthopson) [inline shared/src/business/useCases/workitems/getInboxMessagesForSectionInteractor.js:8]: `getInboxMessagesForSection`

File: shared/src/business/useCases/workitems/getSentMessagesForSectionInteractor.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @param applicationContext
+ * @returns {Promise<*|*>}
+ */
+exports.getSentMessagesForSection = async ({ applicationContext, section }) => {
+  const user = applicationContext.getCurrentUser();
+
+  if (!isAuthorized(user, WORKITEM, user.userId)) {
+    throw new UnauthorizedError(
+      'Unauthorized for getting completed work items',
```
Comment (matthopson) [inline shared/src/business/useCases/workitems/getSentMessagesForSectionInteractor.js:17]: If we're using the "messages" terminology, should we update in the error?

File: shared/src/persistence/dynamo/workitems/getDocumentQCBatchedForSection.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    KeyConditionExpression: '#pk = :pk AND #sk >= :afterDate',
+    applicationContext,
+  });
+
+  return workItems.filter(
+    workItem =>
+      !workItem.isInternal &&
+      !workItem.completedAt &&
+      workItem.section === 'irsBatchSection' &&
```
Comment (matthopson) [inline shared/src/persistence/dynamo/workitems/getDocumentQCBatchedForSection.js]: Could we use the `IRS_BATCH_SYSTEM_SECTION` constant here?

File: shared/src/persistence/dynamo/workitems/getDocumentQCBatchedForUser.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    KeyConditionExpression: '#pk = :pk AND #sk >= :afterDate',
+    applicationContext,
+  });
+
+  return workItems.filter(
+    workItem =>
+      !workItem.completedAt &&
+      !workItem.isInternal &&
+      workItem.sentByUserId === userId &&
+      workItem.section === 'irsBatchSection' &&
```
Comment (matthopson) [inline shared/src/persistence/dynamo/workitems/getDocumentQCBatchedForUser.js]: Same here re: `IRS_BATCH_SYSTEM_SECTION`

**Review by matthopson — APPROVED** [review]

**Pull request merged**